x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/advanced_step/advanced_step_form.tsx

```diff
@@ -309,31 +415,53 @@ export const AdvancedStepForm: FC<CreateAnalyticsStepProps> = ({
               label={i18n.translate('xpack.ml.dataframe.analytics.create.numTopClassesLabel', {
                 defaultMessage: 'Top classes',
               })}
-              helpText={i18n.translate(
-                'xpack.ml.dataframe.analytics.create.numTopClassesHelpText',
-                {
-                  defaultMessage:
-                    'The number of categories for which the predicted probabilities are reported.',
-                }
-              )}
+              helpText={getTopClassesHelpText(selectedNumTopClasses)}
+              isInvalid={selectedNumTopClasses === 0 || selectedNumTopClassesIsInvalid}
+              error={[
+                ...(selectedNumTopClasses === 0
+                  ? [getZeroClassesMessage(ELASTIC_WEBSITE_URL, DOC_LINK_VERSION)]
+                  : []),
+                ...(selectedNumTopClassesIsInvalid ? [numClassesTypeMessage] : []),
+              ]}
             >
-              <EuiFieldNumber
+              <EuiComboBox
                 aria-label={i18n.translate(
                   'xpack.ml.dataframe.analytics.create.numTopClassesInputAriaLabel',
                   {
                     defaultMessage:
                       'The number of categories for which the predicted probabilities are reported',
                   }
                 )}
+                singleSelection={true}
+                options={numTopClassesOptions}
+                selectedOptions={numTopClassesSelectedOptions}
+                onCreateOption={(input: string, flattenedOptions = []) => {
+                  const normalizedInput = input.trim().toLowerCase();
+
+                  if (!normalizedInput) {
```

Comment:
is this a check for an empty string? if so, is it worth making that explicit?

Reply:
Made explicit in 7e11c9e92fc44363b2433154ab7562bb55822eba